gateway/middleware.go

```diff
@@ -101,7 +54,7 @@ func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
 	}
 }
 
-func setOp(req, op interface{}) error {
+func SetCollectionOps(req, op interface{}) error {
```

Comment:
I see that this method may return error like `testRequest value is not a struct`. I don't like `testRequest` here. It will be good to fix it and something more user friendly here (e.g. replace `testRequest` with `request`)